geode-core/src/test/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceImplTest.java

```diff
@@ -0,0 +1,219 @@
+    String testParam2 = "test2";
+    parameters.add(testParam1);
+    parameters.add(testParam2);
+
+    configService.updateMethodAuthorizer(mockCache, testAuthorizerName, parameters);
+
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(TestMethodAuthorizer.class);
+
+    TestMethodAuthorizer methodAuthorizer =
+        (TestMethodAuthorizer) configService.getMethodAuthorizer();
+    assertThat(methodAuthorizer.getParameters()).isEqualTo(parameters);
+  }
+
+  @Test
+  public void updateMethodAuthorizerDoesNotChangeMethodAuthorizerWhenSecurityIsEnabledAndClassNameIsNull() {
+    when(mockSecurity.isIntegratedSecurity()).thenReturn(true);
+
+    configService.init(mockCache);
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    configService.updateMethodAuthorizer(mockCache, null, new HashSet<>());
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    verify(configService).logError(any(String.class));
+  }
+
+  @Test
+  public void updateMethodAuthorizerDoesNotChangeMethodAuthorizerWhenSecurityIsEnabledAndClassNameIsNotFound() {
+    when(mockSecurity.isIntegratedSecurity()).thenReturn(true);
+
+    configService.init(mockCache);
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    configService.updateMethodAuthorizer(mockCache, "FakeClassName", new HashSet<>());
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    verify(configService).logError(any(String.class));
+  }
+
+  @Test
+  public void updateMethodAuthorizerDoesNotChangeMethodAuthorizerWhenSecurityIsEnabledAndSpecifiedClassHasNoValidConstructor() {
+    when(mockSecurity.isIntegratedSecurity()).thenReturn(true);
+
+    configService.init(mockCache);
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    configService.updateMethodAuthorizer(mockCache, this.getClass().getName(), new HashSet<>());
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    verify(configService).logError(any(String.class));
+  }
+
+  @Test
+  public void updateMethodAuthorizerDoesNotChangeMethodAuthorizerWhenSecurityIsEnabledAndSpecifiedClassCannotBeInstantiated() {
+    when(mockSecurity.isIntegratedSecurity()).thenReturn(true);
+
+    when(mockCache.isClosed()).thenThrow(new RuntimeException("Test exception"));
+
+    configService.init(mockCache);
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    configService.updateMethodAuthorizer(mockCache, TestMethodAuthorizer.class.getName(),
+        new HashSet<>());
+    assertThat(configService.getMethodAuthorizer()).isInstanceOf(RestrictedMethodAuthorizer.class);
+    verify(configService).logError(any(String.class));
+  }
+
+  private Object[] getMethodAuthorizerClasses() {
```

Comment:
Just a question, not a requirement, from an obsessive compulsive type of person... we know this method is used to provide test parameters, can we add @SuppressWarnings("unused") to it and prevent those annoying yellow warning marks from the IDE? :-).

Reply:
Done